### ISSUE TITLE
Maybe a mistake in equation 19.17 and 19.18?

### ISSUE THREAD

**Issue by AndreaPi** [udlbook/udlbook#71]
Equation 19.16 is

![image](https://user-images.githubusercontent.com/8233615/235695476-c35ec50e-3345-4a06-b14d-756cf3ed3f1b.png)

Equation 19.17 is

![image](https://user-images.githubusercontent.com/8233615/235696084-54af8baa-f6b4-45c8-a2f5-bd0bf3c3c685.png)

Is this correct? Shouldn't eq. 19.17 be obtained from 19.16 by taking the derivative with respect to $\phi$ of the last term in parentheses only (semi-gradient methods, paragraph 9.3 of Sutton & Barto, or see Sergey Levine for the specific case of fitted Q-iteration http://rail.eecs.berkeley.edu/deeprlcourse-fa18/static/slides/lec-7.pdf)? In this case, a -2 is missing in front of $\alpha$. **Note** that since this should be some sort of stochastic gradient descent, it makes sense that there should be a - in front of the gradient term ( \alpha is a positive learning rate as defined in paragraph 19.3.3).

By the way, it would be easier to write issues if you gave us access to the LaTeX formulas...

PS the same error seems to be present in equation 19.18. Worth checking if it appears also in other formulas down the road.

**Comment by udlbook** [comment]
Ah -- you certainly worried me, but I *THINK* that this is correct.  The minus sign before the last term in the top equation cancels with the minus sign from the fact that we are doing gradient descent.  The missing 2 is just absorbed into alpha.  If you look at Levine's slides then the contents of the brackets is the other way around which means it does need a minus sign.   Feel free to discuss further if you think this isn't right.

Hard to give you the latex formulas as they contain a lot of my own definitions, which would need to be extracted as well.

**Comment by udlbook** [comment]
No worries!  I would rather double check these things and am always grateful when someone questions things.
